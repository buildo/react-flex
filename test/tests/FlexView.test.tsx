import { execSync } from "child_process";
import * as fs from "fs";
import * as path from "path";
import * as React from "react";
import * as renderer from "react-test-renderer";
import FlexView from "../../src";

describe("FlexView", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <FlexView vAlignContent="center" hAlignContent="right" grow shrink wrap>
        CONTENT
      </FlexView>
    );
    expect(component).toMatchSnapshot();
  });
});

describe("build", () => {
  it("build script generates every needed file", () => {
    execSync("yarn build");
    expect(
      fs.readdirSync(path.resolve(__dirname, "../../lib"))
    ).toMatchSnapshot();
  });
});
