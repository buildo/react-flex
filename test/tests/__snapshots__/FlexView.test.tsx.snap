// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexView renders correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "alignItems": "center",
      "boxSizing": "border-box",
      "display": "flex",
      "flex": "1 1 auto",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": undefined,
      "justifyContent": "flex-end",
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "minHeight": 0,
      "minWidth": 0,
      "width": undefined,
    }
  }
>
  CONTENT
</div>
`;

exports[`FlexView renders custom intrinsic elements 1`] = `
<form
  className={undefined}
  style={
    Object {
      "alignItems": "center",
      "boxSizing": "border-box",
      "display": "flex",
      "flex": "1 1 auto",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": undefined,
      "justifyContent": "flex-end",
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "minHeight": 0,
      "minWidth": 0,
      "width": undefined,
    }
  }
>
  CONTENT
</form>
`;

exports[`build build script generates every needed file 1`] = `
Array [
  "FlexView.d.ts",
  "FlexView.js",
  "index.d.ts",
  "index.js",
]
`;
